core/app/models/spree/payment_method.rb

```diff
@@ -79,9 +79,7 @@ def available_for_order?(_order)
     end
 
     def available_for_store?(store)
-      return true if store.blank? || store_id.blank?
-
-      store_id == store.id
+      return true if store_ids.empty? || store_ids.include?(store.id)
```

Comment:
```suggestion
      return true if store_ids.include?(store.id)
```

let's only use the new proper behaviour, that leave blank and apply to all stores is hard to understand for people :)